examples/prometheus/src/main/java/io/opentelemetry/example/PrometheusExample.java

```diff
@@ -0,0 +1,80 @@
+package io.opentelemetry.example;
+
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.exporters.prometheus.PrometheusCollector;
+import io.opentelemetry.metrics.AsynchronousInstrument;
+import io.opentelemetry.metrics.LongValueObserver;
+import io.opentelemetry.metrics.Meter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.metrics.MeterSdkProvider;
+import io.prometheus.client.exporter.HTTPServer;
+import java.io.IOException;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Example of using the {@link PrometheusCollector} to convert OTel metrics to Prometheus format and
+ * expose these to a Prometheus instance via a {@link HTTPServer} exporter.
+ *
+ * <p>A {@link LongValueObserver} is used to periodically measure how many incoming messages are
+ * awaiting processing. The {@link LongValueObserver.Callback} gets executed every collection
+ * interval.
+ */
+public class PrometheusExample {
+
+  private final MeterSdkProvider meterSdkProvider = OpenTelemetrySdk.getMeterProvider();
+  private final Meter meter = meterSdkProvider.get("PrometheusExample", "0.7");
+  private final HTTPServer server;
+  private long incomingMessageCount;
+
+  public PrometheusExample(int port) throws IOException {
+
+    LongValueObserver observer =
+        meter
+            .longValueObserverBuilder("incoming.messages")
+            .setDescription("No of incoming messages awaiting processing")
+            .setUnit("message")
+            .build();
+
+    observer.setCallback(
+        new LongValueObserver.Callback<LongValueObserver.LongResult>() {
+          @Override
+          public void update(AsynchronousInstrument.LongResult result) {
+            result.observe(incomingMessageCount, Labels.empty());
+          }
+        });
+
+    PrometheusCollector.newBuilder()
+        .setMetricProducer(meterSdkProvider.getMetricProducer())
+        .buildAndRegister();
+
+    server = new HTTPServer(port);
+  }
+
+  void shutdown() {
+    server.stop();
+  }
+
+  void simulate() {
+    for (int i = 300; i > 0; i--) {
+      try {
+        System.out.println(
+            i + " Iterations to go, current incomingMessageCount is:  " + incomingMessageCount);
+        incomingMessageCount = ThreadLocalRandom.current().nextLong(100);
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+      }
+    }
+  }
+
+  public static void main(String[] args) throws IOException {
+
+    int port = Integer.parseInt(args[0]);
```

Comment:
I would add a simple check that verifies the port is passed as an argument and prints a message if it is missing.

Reply:
Done!

Reply:
Thank you :)